lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -182,6 +210,37 @@ def ensure_present(self):
         else:
             self.create()
 
+    def get_arn(self):
+        region, ec2_url, aws_connect_params = get_aws_connection_info(self.module, boto3=True)
+        if region:
+            client = boto3_conn(self.module, conn_type='client', resource='rds',
+                                region=region, endpoint=ec2_url, **aws_connect_params)
+        else:
+            self.module.fail_json(msg="region must be specified")
+        instance_counts = {}
+        response = client.describe_db_instances()
+        user_id = response['DBInstances'][0]['DBInstanceArn'].split(':')[4]
```

Comment:
Argh, it's so frustrating that the Elasticache API doesn't return ARNs!

This is probably the best way to get user ID though:
https://stackoverflow.com/a/10198870/3538079

There is always a chance that a user allowed to use Elasticache isn't allowed to use `rds.describe_db_instances`, so it's best to rely on as few other APIs as possible (I think everyone has permission to run `iam.get_caller_identity`)

Reply:
cool, thanks